testgrid/tgrun/pkg/runner/vmi/embed/secondarynodecmd.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+
+set -x
+
+function green()
+{
+  text="${1:-}"
+  echo -e "\033[32m$text\033[0m"
+}
+
+function install_jq()
+{
+  echo "$OS_NAME"
+  if [ "$OS_NAME" = "ubuntu" ]; then
+    sudo apt-get install -y jq
+  elif [ "$OS_NAME" = "CentOS" ]; then
+    sudo yum install epel-release -y
+    sudo yum update -y
+    sudo yum install -y jq
+  elif [ "$OS_NAME" = "Oracle Linux" ]; then
+    sudo yum install -y jq
+  else
+    wget -O jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64
+    chmod +x ./jq
+    cp jq /usr/bin
+  fi
+
+  #echo "jq is already installed"
+  # JQ=/usr/bin/jq
+  # curl https://stedolan.github.io/jq/download/linux64/jq > $JQ && chmod +x $JQ
+  # ls -la $JQ
+}
+function get_join_command()
+{
+  joinCommand=$(curl -X GET "$TESTGRID_APIENDPOINT/v1/instance/$TEST_ID/join-command")
+  echo "${joinCommand}" 
+  #echo "$joincommand" | sed 's/{.*secondaryJoin":"*\([0-9a-zA-Z]*\)"*,*.*}/\1/'
+}
+function wait_for_join_commandready()
+{
+  i=0
+  while true
+    do
+        joinCommand=$(get_join_command)
+        curlCommand=$(echo "${joinCommand}" | jq -r '.curlCommand')
+        if [ "$curlCommand" != "" ]; then
+            echo "$joinCommand"
+            break
+        fi
+        i=$((i+1))
+        if [ $i -gt 10 ]; then
+            report_status_update "failed"
+            send_logs
+            exit 0
+        fi
+        sleep 120
+    done
+  echo 
+}
+function remove_first_element()
+{
+  local list=("$@")
+  local rest_of_list=("${list[@]:1}")
+  echo "${rest_of_list[@]}"
+}
+function runJoinCommand() 
+{
+  joinCommand=$(get_join_command)
```

Comment:
Not able to do that!
when I run the command from the variable it will have a pipe (|) which is not recognized somehow when I just run the command from the variable, so I have to divide the command into multiple variables and feed the actual command with these variables.

Reply:
did you try something like?

```bash
eval "$(get_join_command)"
```

Reply:
I'm curious what the error is. if its truly the pipe then why not just split the command into the url and the rest. download the script using `curl -L -o install.sh https://...` and then run `sudo bash install.sh [rest]`?

Reply:
The purpose is not to remove jq. I have no issue with jq. my goal is to avoid parsing the command and to run the command as is. parsing the arguments seems specific (as the arguments are not guaranteed the same from installer to installer) and not future proof (say an argument is added in the future).